frontend/app/utils/open-payments.server.ts

```diff
@@ -1,20 +1,81 @@
 import {
-  type AuthenticatedClient,
   type PendingGrant,
   type WalletAddress,
-  createAuthenticatedClient,
+  type AuthenticatedClient,
   isFinalizedGrant,
-  isPendingGrant
+  isPendingGrant,
+  createAuthenticatedClient
 } from '@interledger/open-payments'
 import { createId } from '@paralleldrive/cuid2'
 import { toWalletAddressUrl } from './utils.server'
+import { createContentDigestHeader } from 'httpbis-digest-headers'
+import { signMessage } from 'http-message-signatures/lib/httpbis'
+import type { Request } from 'http-message-signatures'
+import * as ed from '@noble/ed25519'
+
+interface RequestLike extends Request {
+  body?: string
+}
+
+interface SignOptions {
+  request: RequestLike
+  privateKey: Uint8Array
+  keyId: string
+}
+
+export interface SignatureHeaders {
+  'Signature': string
+  'Signature-Input': string
+}
+
+interface ContentHeaders {
+  'Content-Digest': string
+  'Content-Length': string
+  'Content-Type': string
+}
+type Headers = SignatureHeaders & Partial<ContentHeaders>
 
 async function createClient(env: Env) {
```

Comment:
Maybe also add a comment that it's based on extension code (along with link to code)